crates/uv-console/src/lib.rs

```diff
@@ -72,3 +73,208 @@ pub fn confirm(message: &str, term: &Term, default: bool) -> std::io::Result<boo
 
     Ok(response)
 }
+
+/// Prompt the user for password in the given [`Term`].
+///
+/// This is a slimmed-down version of `dialoguer::Password`.
+pub fn password(prompt: &str, term: &Term) -> std::io::Result<String> {
+    loop {
+        term.write_str(prompt)?;
+        term.show_cursor()?;
+        term.flush()?;
+
+        let input = term.read_secure_line()?;
+
+        term.clear_line()?;
+
+        if !input.is_empty() {
+            return Ok(input);
+        }
+    }
+}
+
+/// Prompt the user for input text in the given [`Term`].
+///
+/// This is a slimmed-down version of `dialoguer::Input`.
+#[allow(
+    // Suppress Clippy lints triggered by `dialoguer::Input`.
+    clippy::cast_possible_truncation,
+    clippy::cast_possible_wrap,
+    clippy::cast_sign_loss
+)]
+pub fn input(prompt: &str, term: &Term) -> std::io::Result<String> {
```

Comment:
Currently, the username allows an empty value, while the password field doesn't, we should use one behavior for both

Reply:
I've gone with allowing empty values in both, since that's a bit more flexible. Let me know if you prefer it the other way

Reply:
+:+1: